### ISSUE TITLE
Issues when fetching from MrDLib

### ISSUE THREAD

**Issue by koppor** [JabRef/jabref#2733]
I wanted to demonstrate the feature on following entry

```
@Manual{TOSCA,
  title =        {Topology and Orchestration Specification for Cloud Applications Version 1.0},
  year =         {2012},
  organization = {OASIS},
  month =        {10},
}
```

Other entries work fine :)

In the concrete case:

Invalid byte 1 of 1-byte UTF-8 sequence.
com.sun.org.apache.xerces.internal.impl.io.MalformedByteSequenceException: Invalid byte 1 of 1-byte UTF-8 sequence.

```
Invalid byte 1 of 1-byte UTF-8 sequence.
com.sun.org.apache.xerces.internal.impl.io.MalformedByteSequenceException: Invalid byte 1 of 1-byte UTF-8 sequence.
	at com.sun.org.apache.xerces.internal.impl.io.UTF8Reader.invalidByte(Unknown Source)
	at com.sun.org.apache.xerces.internal.impl.io.UTF8Reader.read(Unknown Source)
	at com.sun.org.apache.xerces.internal.impl.XMLEntityScanner.load(Unknown Source)
	at com.sun.org.apache.xerces.internal.impl.XMLEntityScanner.scanData(Unknown Source)
	at com.sun.org.apache.xerces.internal.impl.XMLDocumentFragmentScannerImpl.scanCDATASection(Unknown Source)
	at com.sun.org.apache.xerces.internal.impl.XMLDocumentFragmentScannerImpl$FragmentContentDriver.next(Unknown Source)
	at com.sun.org.apache.xerces.internal.impl.XMLDocumentScannerImpl.next(Unknown Source)
	at com.sun.org.apache.xerces.internal.impl.XMLDocumentFragmentScannerImpl.scanDocument(Unknown Source)
	at com.sun.org.apache.xerces.internal.parsers.XML11Configuration.parse(Unknown Source)
	at com.sun.org.apache.xerces.internal.parsers.XML11Configuration.parse(Unknown Source)
	at com.sun.org.apache.xerces.internal.parsers.XMLParser.parse(Unknown Source)
	at com.sun.org.apache.xerces.internal.parsers.AbstractSAXParser.parse(Unknown Source)
	at com.sun.org.apache.xerces.internal.jaxp.SAXParserImpl$JAXPSAXParser.parse(Unknown Source)
	at com.sun.org.apache.xerces.internal.jaxp.SAXParserImpl.parse(Unknown Source)
	at javax.xml.parsers.SAXParser.parse(Unknown Source)
	at org.jabref.logic.importer.fileformat.MrDLibImporter.isRecognizedFormat(MrDLibImporter.java:54)
	at org.jabref.logic.importer.fetcher.MrDLibFetcher.performSearch(MrDLibFetcher.java:58)
	at org.jabref.gui.entryeditor.EntryEditorTabRelatedArticles$MrDLibFetcherWorker.doInBackground(EntryEditorTabRelatedArticles.java:176)
	at org.jabref.gui.entryeditor.EntryEditorTabRelatedArticles$MrDLibFetcherWorker.doInBackground(EntryEditorTabRelatedArticles.java:163)
	at javax.swing.SwingWorker$1.call(Unknown Source)
	at java.util.concurrent.FutureTask.run(Unknown Source)
	at javax.swing.SwingWorker.run(Unknown Source)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
	at java.lang.Thread.run(Unknown Source)

```

**Comment by Siedlerchr** [comment]
Can not reproduce this issue. Maybe a temporary wrong response from the server?

**Comment by koppor** [comment]
Same query string https://api.mr-dlib.org/v1/documents/Topology%20and%20Orchestration%20Specification%20for%20Cloud%20Applications%20Version%201.0/related_documents?partner_id=jabref&app_id=jabref_desktop&app_version=4.0.0-dev--snapshot--2017-03-24--iaas--2d55cda0c&app_lang=en ?

**Comment by Joeran** [comment]
We have recently discovered also a few other issues. for instance, the special charater %27 in the URL causes an error https://api.mr-dlib.org/v1/documents/test%27/related_documents However, this problem is fixed on our dev server https://api-dev.mr-dlib.org/v1/documents/test%27/related_documents

I can reproduce the problem that @koppor  reported
![image](https://cloud.githubusercontent.com/assets/2410381/25013847/89f0a3cc-2075-11e7-8969-ec6ae7e8cbf1.png)
I dont know if this problem maybe wont occur with the latest (development) version. i suggest, we wait until we deployed our development version to the production API (mid of next week). if the problem reported by @koppor  still exists, we will have a closer look.
